source_control/git.py

```diff
@@ -832,9 +840,21 @@ def main():
     git_path  = module.params['executable'] or module.get_bin_path('git', True)
     key_file  = module.params['key_file']
     ssh_opts  = module.params['ssh_opts']
+    umask  = module.params['umask']
 
     result = dict( warnings=list() )
 
+    # evaluate and set the umask before doing anything else
+    if umask != None:
+        if type(umask) != str:
```

Comment:
from ansible.module_utils.six import string_types
[..]

if not isinstance(umask, string_types):